### ISSUE TITLE
fail to converge for task distillation

### ISSUE THREAD

**Issue by XImenau07** [twinkle0331/Xcompression#1]
09/20/2022 21:46:49 - INFO - __main__ -   ***** Running teacher evaluation *****
09/20/2022 21:46:49 - INFO - __main__ -     Task name = RTE
09/20/2022 21:46:49 - INFO - __main__ -     Num examples = 277
09/20/2022 21:46:49 - INFO - __main__ -     Batch size = 32

09/20/2022 21:46:57 - INFO - __main__ -   ***** Eval results *****
09/20/2022 21:46:57 - INFO - __main__ -     acc = 0.6678700361010831
09/20/2022 21:46:57 - INFO - __main__ -     eval_loss = 0.8153898749086592
09/20/2022 21:46:57 - INFO - __main__ -   ***** Running student evaluation *****
  Task name =  RTE
09/20/2022 21:46:57 - INFO - __main__ -     Num examples = 277
09/20/2022 21:46:57 - INFO - __main__ -     Batch size = 32

09/20/2022 21:46:58 - INFO - __main__ -   ***** Eval results *****
09/20/2022 21:46:58 - INFO - __main__ -     Task name = RTE
09/20/2022 21:46:58 - INFO - __main__ -     acc = 0.5270758122743683
09/20/2022 21:46:58 - INFO - __main__ -     eval_loss = 0.6914981736077203

09/20/2022 21:46:58 - INFO - __main__ -   ***** Running training *****
09/20/2022 21:46:58 - INFO - __main__ -     Task name = RTE
09/20/2022 21:46:58 - INFO - __main__ -     Num examples = 2490
09/20/2022 21:46:58 - INFO - __main__ -     Batch size = 32
09/20/2022 21:46:58 - INFO - __main__ -     Num steps = 1540

FAIL !!! to converge for both w/o pred_distill and w/ pred_distill

**Comment by XImenau07** [comment]
Can offer me a bert-base-72-384 gd file ? I tried to finetune bert-base-72-384 after gd with 3 epochs on RTE task, but failed to get the accuracy metioned in your works. By contrast, i tried to finetune bert-base with 3 epochs on RTE task and it achieve ideal accuracy

**Comment by twinkle0331** [comment]
Hello, gd means general distill. Is your bert-base-72-384 checkpoint trained with general distillation on pre-training corpus? Also bert-base-72-384 is trained with task-specific distillation and the data augmentation follows TinyBERT(https://arxiv.org/abs/1909.10351). I guess you only tries to fine-tune on downstream tasks.

**Comment by twinkle0331** [comment]
As seen in tab. 7 in our paper, fine-training w/o KD means only training on downstream tasks without KD. It even perform worser than your reported result((training on downstream tasks with KD).  As referred in our paper, the decomposition may result in a discrepancy between the raw weights and approximated weights, and exact decomposition is impossible with large compression ratios. So we introduce GD+TD to recover the performance.  Since the downstream task is relatively small, training only on downstream task fails to simulate the prediction of raw model. I close the issue, feel free to reopen it if you have further questions.
